WooCommerce/Classes/ViewModels/Refunds/RefundedProductsDataSource.swift

```diff
@@ -0,0 +1,195 @@
+    ///
+    func update(order: Order) {
+        self.order = order
+    }
+
+    /// Products from a Refund
+    ///
+    var products: [Product] {
+        return resultsControllers.products
+    }
+}
+
+
+// MARK: - Conformance to UITableViewDataSource
+//
+extension RefundedProductsDataSource: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = sections[indexPath.section].rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, for: row, at: indexPath)
+
+        return cell
+    }
+}
+
+
+// MARK: - Support for UITableViewDelegate
+//
+extension RefundedProductsDataSource {
+    /// Set up table section headings
+    ///
+    func viewForHeaderInSection(_ section: Int, tableView: UITableView) -> UIView? {
+        guard let leftText = sections[section].title else {
+            return nil
+        }
+
+        let headerID = TwoColumnSectionHeaderView.reuseIdentifier
+        guard let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: headerID) as? TwoColumnSectionHeaderView else {
+            fatalError()
+        }
+
+        headerView.leftText = leftText
+        headerView.rightText = sections[section].rightTitle
+
+        return headerView
+    }
+}
+
+
+// MARK: - Support for UITableViewDataSource
+//
+private extension RefundedProductsDataSource {
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as ProductDetailsTableViewCell where row == .orderItemRefunded:
+            configureOrderItemRefund(cell, at: indexPath)
+        default:
+            fatalError("Unidentified customer info row type")
+        }
+    }
+
+    /// Setup: Refunded product details cell
+    ///
+    func configureOrderItemRefund(_ cell: ProductDetailsTableViewCell, at indexPath: IndexPath) {
+        let item = items[indexPath.row]
+        let product = lookUpProduct(by: item.productID)
+        let itemViewModel = OrderItemRefundViewModel(item: item,
+                                                     currency: order.currency,
+                                                     product: product)
+        let imageService = ServiceLocator.imageService
+
+        cell.selectionStyle = .default
+        cell.configure(item: itemViewModel, imageService: imageService)
+    }
+}
+
+
+// MARK: - Lookup products
+//
+private extension RefundedProductsDataSource {
+    func lookUpProduct(by productID: Int64) -> Product? {
+        return products.filter({ $0.productID == productID }).first
```

Comment:
nit: maybe you can use `products.first(where: { $0.productID == productID })`

Reply:
🤔  `[filter](https://developer.apple.com/documentation/swift/array/3392999-filter)` is the newer method compared to `[first(where:)](https://developer.apple.com/documentation/swift/array/1848165-first)` , and both are O(n) complexity. Is there something that makes you like `first(where:)` better?